src/api/app/controllers/webui/user_controller.rb

```diff
@@ -193,28 +193,22 @@ def change_password
       return
     end
 
-    # check the valid of the params
-    unless User.current.password_equals?(params[:password])
-      errmsg = 'The value of current password does not match your current password. Please enter the password and try again.'
-    end
-    if params[:new_password] != params[:repeat_password]
-      errmsg = 'The passwords do not match, please try again.'
-    end
-    if params[:password] == params[:new_password]
-      errmsg = 'The new password is the same as your current password. Please enter a new password.'
-    end
-    if errmsg
-      flash[:error] = errmsg
-      redirect_to action: :show, user: User.current
-      return
-    end
-
     user = User.current
-    user.update_password params[:new_password]
-    user.save!
 
-    flash[:success] = 'Your password has been changed successfully.'
-    redirect_to action: :show, user: User.current
+    if user.authenticate(params[:password])
+      user.password = params[:new_password]
+
```

Comment:
What happened to the **repeat_password** param? isn't needed to check if it is the same as the **new_password** one?

Reply:
I dropped it because it's useless. I don't see any point in having it.

Reply:
Is not mandatory in the UI? I thought that the user must repeat the password always is going to change it... 🤔